enterprise/core-edge/src/main/java/org/neo4j/coreedge/catchup/storecopy/edge/StoreCopyClient.java

```diff
@@ -58,4 +60,19 @@ public long copyStoreFiles( CoreMember from, StoreFileStreams storeFileStreams )
             coreClient.removeStoreFileStreamingCompleteListener( fileStreamingCompleteListener );
         }
     }
+
+    public StoreId fetchStoreId( CoreMember from ) throws StoreIdDownloadFailedException
+    {
+        CompletableFuture<StoreId> storeIdCompletableFuture = new CompletableFuture<>();
+        coreClient.setStoreIdConsumer( storeIdCompletableFuture::complete );
+        coreClient.requestStoreId( from );
+        try
+        {
+            return storeIdCompletableFuture.get();
```

Comment:
Are you sure? I do not think it is ok to have unbounded wait anywhere because if some issue occurs, thread should not just hang infinitely.